### PR TITLE
Update function to get all hosts on the subnet

### PR DESCRIPTION
**Pull request by jonasbostoen**
I thought this way of finding all hosts on the network was a little more intuitive.

### CODE REVIEW

**Comment by jonasbostoen** [comment]
@googlebot I signed it!

File: examples/arpscan/arpscan.go [hunk 1]
```diff
@@ -176,13 +176,13 @@ func writeARP(handle *pcap.Handle, iface *net.Interface, addr *net.IPNet) error
...
 	num := binary.BigEndian.Uint32([]byte(n.IP))
 	mask := binary.BigEndian.Uint32([]byte(n.Mask))
-	num &= mask
-	for mask < 0xffffffff {
+	net := num & mask
+	broadcast := net | ^mask
+	for net++; net < broadcast-1; {
 		var buf [4]byte
-		binary.BigEndian.PutUint32(buf[:], num)
+		net++
```
Comment (gconnell) [inline examples/arpscan/arpscan.go:183]: rather than this, how abut make line 181 be:

for net++; net<broadcast; net++ {
  ...
}

Comment (jonasbostoen) [reply examples/arpscan/arpscan.go]: Right! Missed that completely, thanks.

File: examples/arpscan/arpscan.go [hunk 2]
```diff
@@ -178,9 +178,8 @@ func ips(n *net.IPNet) (out []net.IP) {
 	mask := binary.BigEndian.Uint32([]byte(n.Mask))
 	net := num & mask
 	broadcast := net | ^mask
-	for net++; net < broadcast-1; {
+	for net++; net < broadcast-1; net++ {
```
Comment (gconnell) [inline examples/arpscan/arpscan.go:181]: given this new approach, I don't think you want "broadcast - 1" anymore; you won't hit .254 in a /24 with this code.

Comment (jonasbostoen) [reply examples/arpscan/arpscan.go]: Fixed it!

**Comment by gconnell** [comment]
THanks!

**Pull request merged**